### PR TITLE
ENYO-1492: Allow for dynamic changing of the enableBackHistoryAPI property

### PR DESCRIPTION
**Pull request by aarontam**
### Issue
There are cases where we would like to dynamically enable/disable the back history API. Currently this is set to a default of `true` and is updated accordingly based on an application flag.

### Fix
We implement a change handler for the `enableBackHistoryAPI` property.

Enyo-DCO-1.1-Signed-off-by: Aaron Tam <aaron.tam@lge.com>

### CODE REVIEW

**Comment by DavidUM** [comment]
It looks good. I didn't consider this one. 
Thank you @aarontam

File: lib/History/History.js [hunk 1]
```diff
@@ -158,6 +159,17 @@ var History = module.exports = kind.singleton({
 	},
 
 	/**
+	* @private
+	*/
+	enableBackHistoryAPIChanged: function () {
+		if (this.enableBackHistoryAPI) {
+			window.onpopstate = this.bindSafely(function() {this.popStateHandler();});
```
Comment (ryanjduffy) [inline lib/History/History.js]: I don't think we want to assign the event handler like this. `onpopstate` is already included in the enyo dispatcher so we could use Signals here. We could also use `dispatcher.listen()` to add the listener. Either would be okay, imo.

Comment (aarontam) [inline lib/History/History.js]: Ah ha yes good call, updating. :)

File: lib/History/History.js [hunk 2]
```diff
@@ -89,6 +96,12 @@ var History = module.exports = kind.singleton({
 		if (this.enableBackHistoryAPI) {
 			this._initHistoryState();
 		}
+
+		this._popStateHandler = this.bindSafely(function () {
```
Comment (ryanjduffy) [inline lib/History/History.js]: Not a big deal but there's no need to wrap `popStateHandler` with another function. `this.bindSafely(this.popStateHandler)` should do the trick, right?

Comment (aarontam) [inline lib/History/History.js]: Whoops, holdover from the original change, updated!

**Pull request merged**